### ISSUE TITLE
MS SQL script with variable and ;WITH

### ISSUE THREAD

**Issue by vpetryaev** [dbeaver/dbeaver#21954]
### Description

DBEaver-ce 23.2.5 not recognized MS SQL variables after ;WITH
In 23.2.4 - all OK.

### DBeaver Version

Community Edition Version 23.2.5.202311191730

### Operating System

Ubuntu 20.04.6 LTS

### Database and driver

Microsoft SQL Server 2017 (RTM-CU31-GDR) (KB5021126) - 14.0.3460.9 (X64) 
	Jan 25 2023 08:42:43 
	Copyright (C) 2017 Microsoft Corporation
	Developer Edition (64-bit) on Windows Server 2016 Standard 10.0 <X64> (Build 14393: ) (Hypervisor)

com.microsoft.sqlserver mssql-jdbc 12.2.0.jre8

### Steps to reproduce

Run this script on MS SQL:

DECLARE @Var int = 5
;WITH CTE AS (
  SELECT 1 AS Val
  UNION ALL
  SELECT 5 AS Val )
SELECT * FROM CTE WHERE Val<@Var

Result:
SQL Error [137] [S0002]: Must declare the scalar variable "@Var".

**Comment by passacaglia** [comment]
It's fixed in the early access version https://dbeaver.io/files/ea/

**Comment by E1izabeth** [comment]
It's a duplicate of #21898
The bug has been already fixed.

**Comment by perverez** [comment]
I have the same problem with a similar script. Installing early access version on version 23.3.3.202401120022 did not help.

```
DECLARE @one INT = 1
;WITH CTE AS (
  SELECT @one AS 'one'
)
SELECT * FROM CTE
```

**Comment by E1izabeth** [comment]
You can select the whole query and press `Execute SQL Query`
![image](https://github.com/dbeaver/dbeaver/assets/28875055/722f10df-970c-4130-b6be-af394c7989e1)

Does it help?

**Comment by E1izabeth** [comment]
I will reopen this issue as we want to improve support of such queries execution

**Comment by perverez** [comment]
Whole query has the same problem (Must declare the scalar variable "@one"). Before posting the comment I tried all the variations of running the script. When running the whole query the error is displayed in a new tab "Statistics 1", when running the Execute SQL script the error is displayed as a MessageBox.

**Comment by E1izabeth** [comment]
Unfortunately, I can't reproduce it



https://github.com/dbeaver/dbeaver/assets/28875055/7b387861-36ef-41e2-9ccf-73eb2e623d1f

**Comment by Tix2garr** [comment]
It works fine for me, _however_, I have made changes in the SQL Processing > Delimiters area.

Statements delimiter:  GO
X  Ignore native delimiter
X  Blank line is statement delimiter
X  Remove trailing query delimiter

If you change these to the defaults for MS SQL, it will give that error.
Statements delimiter:  ;
    Ignore native delimiter
X  Blank line is statement delimiter
X  Remove trailing query delimiter

Michael

**Comment by perverez** [comment]
Yes, I confirm, if I did the setup according to Michael (which is of course not the solution to the problem), the script processed correctly.
